src/providers/Context/Context.provider.jsx

```diff
@@ -13,29 +15,59 @@ function useVideosUpdate() {
   return useContext(VideoUpdateContext);
 }
 
+function useVideoDetailed() {
+  return useContext(VideoDetailedContext);
+}
+
+function reducer(state, action) {
+  switch (action.type) {
+    case 'grid':
+      return { videos: action.payload.data, video: null };
+    case 'detail':
+      return { videos: null, video: action.payload.data };
+    default:
```

Comment:
Just as a comment, here you can see that action holds the type and the payload for that action, but having payload key name at every action seems cumbersome for me since you can provide data or the video directly reducing the boilerplate code